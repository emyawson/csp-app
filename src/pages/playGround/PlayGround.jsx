import React from 'react';
import BCrumb from '../../components/Bcrumb/BCrumb.jsx';
import Layout from '../../components/layout/Layout.jsx';
import Survey from '../../components/survey/Survey.jsx';

const PlayGround = () => (
        <Layout>
            <BCrumb />
            <Survey />
        </Layout>
);

export default PlayGround;
