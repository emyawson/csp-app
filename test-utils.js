import React from 'react';
import { Provider } from 'react-redux';
import { createStore, combineReducers } from 'redux';
import { Router } from 'react-router-dom';
import { connectRouter } from 'connected-react-router';
import { render } from '@testing-library/react';
import { persistReducer } from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import autoMergeLevel2 from 'redux-persist/lib/stateReconciler/autoMergeLevel2';
import { history } from './src/store';
import reducers from './src/store/reducer/index';

const createRootReducer = (reducerHistory) => combineReducers({
  router: connectRouter(reducerHistory),
  ...reducers,
});

// Setup redux persist
const persistConfig = {
  key: 'root',
  storage,
  stateReconciler: autoMergeLevel2,
};
const persistedReducer = persistReducer(
  persistConfig,
  createRootReducer(history),
);

const renderWithRouter = (ui, options = {}, initialState) => ({
  ...render(
      <Provider store={createStore(persistedReducer, initialState)}>
        <Router history={history}>{ui}</Router>
      </Provider>,
  ),
});

const AllTheProviders = () => {};
const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render, renderWithRouter };
