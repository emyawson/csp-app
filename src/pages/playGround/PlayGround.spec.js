import React from 'react';
// import { render, screen } from "@testing-library/react";
import PlayGround from './PlayGround.jsx';
// import { renderWithRouter } from "../test-utils";
import { renderWithRouter } from '../../../test-utils';

const defaultState = {};

const renderComponent = (state = {}) => {
  const combinedState = {
    ...defaultState,
    ...state,
  };
  return renderWithRouter(<PlayGround />, combinedState);
};

test('renders <PlayGround/>', () => {
  const { container } = renderComponent();
  expect(container).toMatchSnapshot();

  // const linkElement = screen.getByText(/learn react/i);
  // expect(linkElement).toBeInTheDocument();
});
