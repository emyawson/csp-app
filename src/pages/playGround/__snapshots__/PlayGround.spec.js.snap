// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <PlayGround/> 1`] = `
<div>
  <main
    class="page-container"
  >
    <div
      class="page-container__inner"
    >
      <div
        class="page-container__main"
      >
        <div
          class="breadcrumb"
        >
          <nav
            aria-label="Breadcrumb"
            class="breadcrumb__nav"
            data-cytest="breadcrumb"
            role="navigation"
          >
            <ol
              class="breadcrumb__list"
              itemprop="breadcrumb"
              itemscope=""
              itemtype="http://schema.org/BreadcrumbList"
            >
              <li
                class="breadcrumb__list-item"
                itemprop="itemListElement"
                itemscope=""
                itemtype="http://schema.org/ListItem"
              >
                 
                <a
                  class="breadcrumb__link link link--black"
                  href="/nl/nl"
                  itemprop="item"
                >
                  
                  <span
                    class="breadcrumb__item"
                    data-cytes="crumb"
                  >
                    IKEA
                  </span>
                </a>
                 
                <meta
                  content="1"
                  itemprop="position"
                />
              </li>
              <li
                class="breadcrumb__list-item"
                itemprop="itemListElement"
                itemscope=""
                itemtype="http://schema.org/ListItem"
              >
                 
                <a
                  class="breadcrumb__link link link--black"
                  href="/nl/nl/customer-service"
                  itemprop="item"
                >
                  
                  <span
                    class="breadcrumb__item"
                    data-cytes="crumb"
                  >
                    Klantenservice
                  </span>
                </a>
                 
                <meta
                  content="2"
                  itemprop="position"
                />
              </li>
              <li
                class="breadcrumb__list-item"
                itemprop="itemListElement"
                itemscope=""
                itemtype="http://schema.org/ListItem"
              >
                 
                <a
                  class="breadcrumb__link link link--black"
                  href="/nl/nl/customer-service/support"
                  itemprop="item"
                >
                  
                  <span
                    class="breadcrumb__item"
                    data-cytes="crumb"
                  />
                </a>
                 
                <meta
                  content="3"
                  itemprop="position"
                />
              </li>
            </ol>
          </nav>
        </div>
        <div
          class="container"
        >
          <div
            class="container__title"
          >
            <h4>
              How was your experience today?
            </h4>
          </div>
          <div
            class="container__images"
          >
            <reactcomponent
              alt="Angry"
              class="container__images__angry"
            />
            <reactcomponent
              alt="Annoyed"
              class="container__images__annoyed"
            />
            <reactcomponent
              alt="Neutral"
              class="container__images__neutral"
            />
            <reactcomponent
              alt="Happy"
              class="container__images__happy"
            />
            <reactcomponent
              alt="Very Happy"
              class="container__images__very-happy"
            />
          </div>
        </div>
      </div>
    </div>
  </main>
</div>
`;
